pkg/ebpf/c/tracee.bpf.c

```diff
@@ -3486,27 +3529,71 @@ int tracepoint__sched__sched_process_exec(struct bpf_raw_tracepoint_args *ctx)
     // We passed all filters (in should_trace()) - add this pid to traced pids set
     data.task_info->follow = true;
 
-    struct task_struct *task = (struct task_struct *) ctx->args[0];
-    struct linux_binprm *bprm = (struct linux_binprm *) ctx->args[2];
+    if (should_submit(SCHED_PROCESS_EXEC, data.config) || data.config->options & OPT_PROCESS_INFO) {
```

Comment:
Now that you moved everything under that `if`, it will make more sense to check the negation of this condition and return if true. This will save the extra indentation.